### ISSUE TITLE
Props destructuring with `...rest` spread reports type error

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#68]
`svelte-check-rs` reports false positive errors when using props destructuring with rest spread (`...rest`).

## Reproduction

```svelte
<script lang="ts">
  import type { HTMLAttributes } from 'svelte/elements'

  let { element, missing_msg = `No image`, ...rest }:
    & HTMLAttributes<HTMLImageElement | HTMLDivElement>
    & { element: { name: string }; missing_msg?: string } = $props()
</script>

<div {...rest}>
  {element.name}
</div>
```

## Error

```
Error: Type 'Record<string, unknown>' is not assignable to type 'HTMLAttributes<HTMLDivElement> & { element: { name: string }; missing_msg?: string | undefined; } & Record<...>'. (ts(TS2322))
```

## Expected

No error. This is valid Svelte 5 props destructuring with rest spread.

## Environment

- svelte-check-rs: 0.6.5
- svelte: 5.46.1
